### PR TITLE
Make our custom element implementation directly consumable by CE V1

### PR DESCRIPTION
**Pull request by cramforce**
The CustomElement V1 API is strictly coupled to ES6 classes inheriting from build-in `HTMLElement` or its subclasses. Alternatively, `Reflect.construct` can be used to make legacy constructors inherit from build-in classes.

Because we transpile ES6 classes, we never hit any of those 2 paths (Not an ES6 class and not using `Reflect.construct` in transpiled output). This is hidden by the polyfill by falling back to a V0 polyfill instead. As a consequence we depended on the polyfill even in browsers that implement custom elements.

This CL manually uses a legacy class with a `Reflect.construct` based constructor to make the resulting object directly consumable by `customElements.define`. Note, that while there is a way to make Babel produce similar code, there is no much way for closure compiler transpilation.

Additionally: The CE polyfill had a side effect and was included multiple times. Due to the polyfill's behavior this led to a forced CE V1 polyfill behavior (and it is also bad). We'll try to get this fixed upstream, but for now we always patch the polyfill (we had already done this for the prod build).

### CODE REVIEW

File: src/polyfills.js [hunk 1]
```diff
@@ -16,8 +16,10 @@
 
 // Importing the document-register-element module has the side effect
 // of installing the custom elements polyfill if necessary.
+/*eslint-disable */
```
Comment (erwinmombay) [inline src/polyfills.js]: this should no longer be needed. we now allow imports > 80 chars

Comment (cramforce) [reply src/polyfills.js]: Still needed or is this very new?

**Comment by erwinmombay** [comment]
a few conflicts here (mostly just naming, yay!) to the changes I have in https://github.com/ampproject/amphtml/pull/12558.

I'll wait for this to go in so I can resolve conflicts in #12558

File: src/polyfills.js (hunk 1 above)
Comment (erwinmombay) [reply src/polyfills.js]: @cramforce on rebase you shouldn't need to do the disable / enable dance

**Comment by cramforce** [comment]
Thanks. I might need to work a bit on tests potentially. I'm worried some of them relied on the side effect I took away :)

File: src/polyfills.js [hunk 2]
```diff
@@ -35,7 +37,7 @@ import {getMode} from './mode';
   sure to not `install` it during dev since the `install` is done as a side
   effect in importing the module.
 */
-if (!getMode().localDev) {
+if (!getMode().localDev && !self.customElements) {
```
Comment (erwinmombay) [inline src/polyfills.js]: so this is mostly a runtime path change right. module is still included but document-register-element no longer takes over if custom elements v1 is supported.

Comment (cramforce) [reply src/polyfills.js]: Right. I might take it out, but the polyfill should now do nothing anyway.

**Comment by erwinmombay** [comment]
@cramforce from what I vaguely remember, i think tests did rely on the side effect.

**Comment by erwinmombay** [comment]
@cramforce we could also use the ponyfill version https://github.com/WebReflection/document-register-element#new-ponyfill-in-13

which has no side effects by default

**Comment by cramforce** [comment]
@erwinmombay We can consider switching to that. But for now we actually want the comprehensive feature detection. I'd rather not change that in this patch.

**Comment by cramforce** [comment]
Woot. Got unit tests to pass. Some integration failures remaining.

**Comment by cramforce** [comment]
I tried all day, but so far haven't been able to land this.

What I know:

- An integration test that creates an instagram placeholder fails, because the `amp-img` is never actually activated into a custom element
- Although the direct amp-img on the same page work
- And it is created in the right window
- And amp-img remains registered.
- It fails only in browsers with native support
- It works in the same browser locally in the exact same versions

- Another test that fails is ad related. It appears that upgrading the late loaded elements fails.

**Comment by aghassemi** [comment]
Thanks for the pull request. Since this PR has been stale for a while, I am closing it. Please feel free to reopen as needed.

**Pull request closed without merging**